source/MaterialXTest/ShaderValid.cpp

```diff
@@ -868,6 +872,40 @@ static void runOSLValidation(const std::string& shaderName, mx::TypedElementPtr
 
                 if (testOptions.renderImages)
                 {
+                    // Look for textures and build parameter override string for each image
+                    // files if a relative path maps to an absolute path
+                    const mx::Shader::VariableBlock publicUniforms = 
+                        shader->getUniformBlock(mx::Shader::PIXEL_STAGE, mx::Shader::PUBLIC_UNIFORMS);
+
+                    mx::StringVec overrides;
+                    mx::StringMap separatorMapper;
+                    separatorMapper["\\\\"] = "/";
+                    separatorMapper["\\"] = "/";
+                    for (auto uniform : publicUniforms.variableOrder)
+                    {
+                        if (uniform->type != MaterialX::Type::FILENAME)
+                        {
+                            continue;
+                        }
+                        if (uniform->value)
+                        {
+                            const std::string& uniformName = uniform->name;
+                            mx::FilePath filename;
+                            mx::FilePath origFilename(uniform->value->getValueString());
+                            if (!origFilename.isAbsolute())
```

Comment:
For each input filename try and find the absolute path name if is relative. If found then set parameter overrides for the shader.